src/components/PatientRecord/PatientRecord.tsx

```diff
@@ -14,29 +14,120 @@ import {
   ReportsVisualizer
 } from 'fhir-visualizers';
 import { DomainResource } from 'fhir-objects';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faNotesMedical,
+  faChevronLeft,
+  faChevronUp,
+  faChevronDown
+} from '@fortawesome/free-solid-svg-icons';
+import styles from './PatientRecord.module.scss';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+
 const getResourceByType = (
   patientRecord: ReadonlyArray<DomainResource>,
   resourceType: string
 ): ReadonlyArray<object> => {
   return patientRecord.filter(resource => resource.resourceType === resourceType);
 };
 
+interface PatientRecordElementProps {
+  resourceType: string;
+}
+
 const PatientRecord: FC = () => {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const resourceTypes = [
+    'Patient',
+    'Condition',
+    'Observation',
+    'DiagnosticReport',
+    'MedicationRequest',
+    'AllergyIntolerance',
+    'CarePlan',
+    'Procedure',
+    'Encounter',
+    'Immunization'
+  ];
+
+  const expand = (): void => {
+    setIsExpanded(!isExpanded);
+  };
+
+  // Set the height of the patient record container
+  useEffect(() => {
+    const recordContainer = document.getElementById('recordContainer');
+    const headerHeight = document.getElementById('header')?.clientHeight;
+    const navHeight = document.getElementById('navigation')?.clientHeight;
```

Comment:
Same thing here, are we able to pass refs to these elements through the components?  Feels like that would be the most React-appropriate way to go about doing this.  Though I personally don't see anything wrong with accessing elements like this, I do know its slower and causes some rendering inefficiencies from when I was implementing sticky in IE.  Not really noticeable on chrome but it could cause half a frame of lag on a samsung smart fridge running IE7, for example.

Reply:
Updated to using refs. Do you have a samsung smart fridge running IE7 I can test on?

Reply:
I don't commune with the smart-fridgeless, sorry